pkg/config/config.go

```diff
@@ -160,6 +168,41 @@ func LoadFile(filename string, expandEnvVars bool, c *Config) error {
 	return LoadBytes(buf, expandEnvVars, c)
 }
 
+// LoadRemote reads a config from url
+func LoadRemote(url string, expandEnvVars bool, c *Config) error {
+	remoteOpts := &RemoteOpts{}
+	if c.BasicAuthUser != "" || c.BasicAuthPassFile != "" {
+		remoteOpts.HTTPClientConfig = &config.HTTPClientConfig{
+			BasicAuth: &config.BasicAuth{
+				Username:     c.BasicAuthUser,
+				PasswordFile: c.BasicAuthPassFile,
+			},
+		}
+	}
+
+	if remoteOpts.HTTPClientConfig != nil {
+		dir, err := os.Getwd()
+		if err != nil {
+			return err
+		}
+		remoteOpts.HTTPClientConfig.SetDirectory(dir)
+	}
+
+	rc, err := NewRemoteConfig(url, remoteOpts)
+	if err != nil {
+		return errors.Errorf("error reading remote config: %s", err.Error())
```

Comment:
please wrap the errors instead of using errors.Errorf so they can be unwrapped: `fmt.Errorf("error reading remote config: %w", err)`  (ditto for other places you do this)